thirdparty/github.go

```diff
@@ -541,3 +541,73 @@ func CheckGithubAPILimit(ctx context.Context, oauthToken string) (int64, error)
 
 	return int64(limits.Core.Remaining), nil
 }
+
+// GithubUserInOrganization returns true if the given github user is in the
+// given organization. The user with the attached token must have
+// visibility into organization membership, including private members
+func GithubUserInOrganization(ctx context.Context, token, requiredOrganization, username string) (bool, error) {
+	httpClient, err := getGithubClient(token)
+	if err != nil {
+		return false, errors.Wrap(err, "can't fetch data from github")
+	}
+	defer util.PutHTTPClient(httpClient)
+
+	client := github.NewClient(httpClient)
+
+	// doesn't count against API limits
+	limits, _, err := client.RateLimits(ctx)
+	if err != nil {
+		return false, err
+	}
+	if limits == nil || limits.Core == nil {
+		return false, errors.New("rate limits response was empty")
+	}
+	if limits.Core.Remaining < 3 {
+		return false, errors.New("github rate limit would be exceeded")
+	}
+
+	isMember, _, err := client.Organizations.IsMember(context.Background(), requiredOrganization, username)
+	return isMember, err
+}
+
+// GetPullRequestMergeBase returns the merge base hash for the given PR.
+// This function will retry up to 5 times, regardless of error response (unless
+// error is the result of hitting an api limit)
+func GetPullRequestMergeBase(ctx context.Context, token, owner, repo string, prNumber int) (string, error) {
```

Comment:
I worry about confusions for the three successive string arguents, and might tend to make a type for making this more clear.

Reply:
Given the way this is used, do you think supplying patch.GithubPatchData is appropriate?